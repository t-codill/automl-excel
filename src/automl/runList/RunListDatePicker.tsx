
import { DatePicker, DayOfWeek, IDatePickerStrings } from "office-ui-fabric-react";
import * as React from "react";

interface IRunListDatePickerProps {
    onDateChange(
        startDate: Date | undefined,
        isDateRange: boolean
    ): void;
}

interface IRunListDatePickerState {
    datePickerSelect: Date | undefined;
    firstDayOfWeek: DayOfWeek.Sunday;
}

export const dayPickerStrings: IDatePickerStrings = {
    months: [
        "January",
        "February",
        "March",
        "April",
        "May",
        "June",
        "July",
        "August",
        "September",
        "October",
        "November",
        "December"
    ],
    shortMonths: ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"],
    days: ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"],
    shortDays: ["S", "M", "T", "W", "T", "F", "S"],
    goToToday: "Go to today",
    prevMonthAriaLabel: "Go to previous month",
    nextMonthAriaLabel: "Go to next month",
    prevYearAriaLabel: "Go to previous year",
    nextYearAriaLabel: "Go to next year",
    closeButtonAriaLabel: "Close date picker"
};

export class RunListDatePicker extends React.Component<IRunListDatePickerProps, IRunListDatePickerState
    > {
    constructor(props: IRunListDatePickerProps) {
        super(props);
        this.state = {
            datePickerSelect: undefined,
            firstDayOfWeek: DayOfWeek.Sunday,
        };
    }
    public readonly render = (): React.ReactNode => {
        return (
            <DatePicker
                value={this.state.datePickerSelect}
                firstDayOfWeek={this.state.firstDayOfWeek}
                strings={dayPickerStrings}
                onSelectDate={this.handleDatePickerSelect}
                placeholder="Select a date..."
                ariaLabel="Select a date"
            />
        );
    }
    private readonly handleDatePickerSelect = (date: Date | null | undefined): void => {
        if (date) {
            this.setState({ datePickerSelect: date });
            this.props.onDateChange(date, false);
        }
        return;
    }
}
