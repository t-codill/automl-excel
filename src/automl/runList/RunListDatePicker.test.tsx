import { shallow, ShallowWrapper } from "enzyme";
import { DatePicker, IDatePickerProps } from "office-ui-fabric-react";
import * as React from "react";
import { RunListDatePicker } from "./RunListDatePicker";

describe("RunListDatePicker", () => {
    let tree: ShallowWrapper;
    const mockCallBack = jest.fn();
    const props = {
        onDateChange: mockCallBack
    };
    let datePickerProps: IDatePickerProps;

    beforeEach(() => {
        tree = shallow(
            <RunListDatePicker {...props} />
        );
        datePickerProps = tree.find(DatePicker)
            .props();
        mockCallBack.mockClear();
    });
    describe("Verify DatePicker", () => {
        it("should render date picker", () => {
            expect(tree)
                .toMatchSnapshot();
        });
        it("should not call onSelectDate with invalid select", () => {
            if (datePickerProps && datePickerProps.onSelectDate) {
                datePickerProps.onSelectDate(undefined);
            }
            expect(mockCallBack)
                .toBeCalledTimes(0);
        });
        it("should call onSelectDate with valid select", () => {
            if (datePickerProps && datePickerProps.onSelectDate) {
                datePickerProps.onSelectDate(new Date(2019, 1, 2, 3, 4, 5));
            }
            expect(mockCallBack)
                .toBeCalledTimes(1);
        });
    });
});
