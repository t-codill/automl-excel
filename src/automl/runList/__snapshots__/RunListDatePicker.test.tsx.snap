// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RunListDatePicker Verify DatePicker should render date picker 1`] = `
<StyledDatePickerBase
  ariaLabel="Select a date"
  firstDayOfWeek={0}
  onSelectDate={[Function]}
  placeholder="Select a date..."
  strings={
    Object {
      "closeButtonAriaLabel": "Close date picker",
      "days": Array [
        "Sunday",
        "Monday",
        "Tuesday",
        "Wednesday",
        "Thursday",
        "Friday",
        "Saturday",
      ],
      "goToToday": "Go to today",
      "months": Array [
        "January",
        "February",
        "March",
        "April",
        "May",
        "June",
        "July",
        "August",
        "September",
        "October",
        "November",
        "December",
      ],
      "nextMonthAriaLabel": "Go to next month",
      "nextYearAriaLabel": "Go to next year",
      "prevMonthAriaLabel": "Go to previous month",
      "prevYearAriaLabel": "Go to previous year",
      "shortDays": Array [
        "S",
        "M",
        "T",
        "W",
        "T",
        "F",
        "S",
      ],
      "shortMonths": Array [
        "Jan",
        "Feb",
        "Mar",
        "Apr",
        "May",
        "Jun",
        "Jul",
        "Aug",
        "Sep",
        "Oct",
        "Nov",
        "Dec",
      ],
    }
  }
/>
`;
